theano/tensor/nlinalg.py

```diff
@@ -179,6 +185,10 @@ class ExtractDiag(Op):
     __props__ = ("view",)
 
     def __init__(self, view=False):
+        warnings.warn("DeprecationWarning: theano.tensor.nlinalg.ExtractDiag"
```

Comment:
Really? :open_mouth: How could it be? Any access to x like x[0] or x[:, 0] returns a CudaNDarray? It only work if there is no arithmetic on x right?